api-reference/10 UI Widgets/dxScheduler/1 Configuration/onAppointmentFormOpening.md

```diff
@@ -30,40 +30,69 @@ The form's instance; created only once - when the function is executed for the f
 Model data. Available only if you use Knockout.
 
 ##### field(e.popup): dxPopup
-<!-- Description goes here -->
+The instance of the popup that contains the form.
 
 ---
-The appointment details form contains the [Form](/concepts/05%20Widgets/Form/00%20Overview.md '/Documentation/Guide/Widgets/Form/Overview/') widget whose instance is passed to this function in the **form** field. Use the [widget's API](/api-reference/10%20UI%20Widgets/dxForm '/Documentation/ApiReference/UI_Widgets/dxForm/') to customize the appointment details form.
+The appointment details form is displayed inside a popup. These elements are implemented by the [Form](/Documentation/Guide/Widgets/Form/Overview/) and [Popup](/Documentation/Guide/Widgets/Popup/Overview/) widgets. Their instances are passed to **onAppointmentFormOpening** in the **form** and **popup** fields. Use the [Form API](/Documentation/ApiReference/UI_Widgets/dxForm/Configuration/) and [Popup API](/Documentation/ApiReference/UI_Widgets/dxPopup/Configuration/) to customize the corresponding widget.
 
-The following code shows how to use the **onAppointmentFormOpening** function to customize an existing form item (`startDate`) and add a new form item (`location`). Note that in the latter case, the array of [form items](/Documentation/ApiReference/UI_Widgets/dxForm/Configuration/#items) should be checked to ensure that it does not already contain an item with the same data field.
+The form items are organized into two groups:
+
+<table class="dx-table">
+    <tr>
+        <th>Group name</th>
+        <th>Description</th>
+    </tr>
+    <tr>
+        <td>mainGroup</td>
+        <td>Contains form fields that define main appointment parameters (subject, start and end dates, etc.).</td>
+    </tr>
+    <tr>
+        <td>recurrenceGroup</td>
+        <td>Contains form fields that define appointment recurrence parameters.</td>
+    </tr> 
+</table>
+
+You can add a custom item to any group or create an ungrouped item and display it under the groups, as shown in the following image:
```

Comment:
What do you mean by display an ungrouped item under groups? ("groups" sounds like this ungrouped item is displayed under multiple groups.) Please clarify.

Reply:
This ungrouped item is displayed under two groups.
![Scheduler_onAppointmentFormOpening](https://user-images.githubusercontent.com/66619750/88156652-5ffb3e00-cc12-11ea-9c56-b3be91bd9b81.png)

Reply:
Thank you for the clarification.